### PR TITLE
test: stories for dialogs

### PR DESCRIPTION
**Pull request by codeincontext**
Adds stories for each of our dialogs. Each dialog has a component, but we interact with them through the DialogContents component. I figured it made sense to test through that interface

### Mocking providers

Some of the dialogs need some mocked data from posthog, the demo status, or the demo API call. For the time being I've mocked each of these at the highest point possible, to help with coverage. eg: Mock the tRPC call from the DemoProvider, rather than mocking the DemoProvider itself. These patterns might change in the future as we learn more about storybook and have more tests in place

### MSW (Mock Service Worker)

This is the recommended way to mock network requests in storybook. It seems to work really nicely out of the box

### Snapshots

You can see the story changes [in chromatic](https://www.chromatic.com/review?appId=672908473f629875f0be294f&number=372&type=linked&view=changes)

### CODE REVIEW

File: apps/nextjs/src/components/DialogControl/DialogContents.tsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ const dialogTitlesAndIcons: Record<
     iconName: "warning",
   },
   "demo-interstitial": {
-    title: "Lesson limit reached",
+    title: "Demo lesson limits",
```
Comment (codeincontext) [inline apps/nextjs/src/components/DialogControl/DialogContents.tsx:38]: This is a copy change, as the previous title was inccorect a lot of the time

File: apps/nextjs/src/components/DialogControl/DialogContents.tsx [hunk 2]
```diff
@@ -67,7 +67,13 @@ const DialogContents = ({
...
         title={dialogTitlesAndIcons[dialogWindow].title}
         iconName={dialogTitlesAndIcons[dialogWindow].iconName}
-        hideIcon={dialogWindow === "feedback" || dialogWindow === "share-chat"}
+        iconOverride={
+          dialogWindow === "feedback" || dialogWindow === "share-chat"
+            ? {
+                $display: "none",
+              }
+            : {}
+        }
```
Comment (codeincontext) [inline apps/nextjs/src/components/DialogControl/DialogContents.tsx:76]: `hideIcon` doesn't seem to exist. This seems to do the job

File: apps/nextjs/.storybook/public/mockServiceWorker.js [hunk 3]
```diff
@@ -0,0 +1,295 @@
+/* eslint-disable */
```
Comment (codeincontext) [inline apps/nextjs/.storybook/public/mockServiceWorker.js:1]: This is autogenerated

File: apps/nextjs/src/components/DialogControl/DialogContents.tsx (hunk 1 above)
Comment (codeincontext) [reply apps/nextjs/src/components/DialogControl/DialogContents.tsx:38]: @mikeritson-oak it might be worth checking this

**Review by JBR90 — APPROVED** [review]
LGTM

File: apps/nextjs/src/components/DialogControl/DialogContents.tsx (hunk 2 above)
Comment (mantagen) [reply apps/nextjs/src/components/DialogControl/DialogContents.tsx:76]: You might need to run `npm install` (to get the later version of oak components)

Comment (codeincontext) [reply apps/nextjs/src/components/DialogControl/DialogContents.tsx:76]: oh! I'll have a look

Comment (codeincontext) [reply apps/nextjs/src/components/DialogControl/DialogContents.tsx:76]: Correct! Brownie points for you. Thanks

**Pull request merged**